spec/models/teacher_spec.rb

```diff
@@ -1,11 +1,25 @@
 require 'rails_helper'
 RSpec.describe Teacher, type: :model do
-  describe 'validations' do
+  describe 'presence validations' do
     subject { build(:teacher) }
 
     it { is_expected.to validate_presence_of(:first_name) }
     it { is_expected.to validate_presence_of(:last_name) }
     it { is_expected.to validate_presence_of(:email) }
     it { is_expected.to validate_presence_of(:password) }
   end
+
+  describe 'email uniqueness' do
+    subject { build(:teacher) }
+
+    it { is_expected.to validate_uniqueness_of(:email).case_insensitive }
+  end
+end
+
+describe 'password length validation' do
+  let(:teacher) { Teacher.new(attributes_for(:teacher, password: 'pea')) }
+
+  it 'fails' do
```

Comment:
When the spec fails, the error message will be sth like that:

```
password length validation
  fails
```

We should aim for a bit more descriptive message

```suggestion
  it 'fails for passwords shorter than 8 characters' do
```

There also will be purists telling you to use contexts like

```suggestion
  context "when password length less than 8 characters" do
    it 'is invalid' do
```

but I do prefer to keep it simple :)